### ISSUE TITLE
Absence of 'beeps etc.'

### ISSUE THREAD

**Issue by Jedsters** [FrSkyRC/ETHOS-Feedback-Community#1281]
On Special Funcs we have the option to play a track or a value. It seems odd that there are no beeps etc. 

By contrast opentx has Beep1/2/3, Warn1/2, Cheep, Ratata, Tick, Siren, Ring, SciFi, Robot, Chirp, Tada, Crickt, AlarmClk. 
While I'm not suggesting Ethos should have the same list, it gives an idea of what people use and what sort of sounds can be useful for assorted purposes. 
In particular I'd suggest at least a couple of informational sounds and a couple of warning sounds and a couple of error type sounds as well as a selection of other sounds. 
Although this could be done by the user and sound files, it does seem an omission that sounds aren't included as part of the basic function.

**Comment by bsongis-frsky** [comment]
We could include some sounds on the SD card. But those OpenTX sounds are there for historical reasons, when there was no SD on 8bits AVR!

Please give us some examples of wav sounds you would like to include and we will add them

**Comment by Jedsters** [comment]
Unfortunately the reason behind the request is that I don't have any, which was rather what was behind my request. I suspect if I had what I wanted I wouldn't have raised the issue, although it would still be nice to have 'noises' as a separate category from things which are spoken. 
Maybe someone else has a selection of beeps and warning noises etc. as .wav they can share with all of us?

**Comment by agibson2** [comment]
I don't see a Play Tone available for special functions.  That would at least allow tones to be played which would be useful if that wouldn't take too much to implement.  The nice thing about tones is that they would not be queued by other wav sounds (I assume).

**Comment by agibson2** [comment]
Audio files don't have the same functionality as beeps and such because they preempt other audio playback.  The nice thing about beeps and such on OS's like OpenTX is that they can be played even when audio files are in process of playing.

**Comment by Jedsters** [comment]
Hopefully the attached sound files can be used as part of the OS as @agibson2 suggests.

**Comment by agibson2** [comment]
The beeps aren't sound files in opentx.  Even Lua in OpenTX has the ability to create sounds using frequency settings (not wav files).  They are generated by the OS directly.  Not that your sounds couldn't be useful though.  Not trying to downplay your sound files.

**Comment by Jedsters** [comment]
@bsongis-frsky I notice this is still marked information needed. 
Please see attached, please let me know if there are any problems.

[OpentxSystemSounds.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/10368715/OpentxSystemSounds.zip)

Thanks

**Comment by Jedsters** [comment]
This has come up a few times recently on RCGroups, in particular with being able to play high priority tones without having to wait for other tracks to finish etc.. 
I notice this is still open so hopefully this can be implemented in due course.
Thanks

**Comment by agibson2** [comment]
Bringing this up again as it was mentioned in rcgroups again.  I still think system level tones that don't need to wait for play files to finish playing are very useful for critical things that you want to know when something happens instead of using sound files that must wait for existing audio plays to finish.  This is one of the things that OpenTX had that Ethos does not.  The tones would happen and the audio that was queued would still be playing and not interrupted either.

**Comment by Jedsters** [comment]
@agibson2  I'm never sure how much visibility issues get once they've been closed. Not sure if another issue referencing this one with the attached .zip is needed? Any idea?

**Comment by bsongis-frsky** [comment]
Still waiting for some help on this one, if somebody can prepare a nice sounds pack.
The one above is not very nice!

**Comment by Jedsters** [comment]
@bsongis-frsky  May I ask what is wrong with it - although I'm not sure if I can improve it?

**Comment by bsongis-frsky** [comment]
I find it a little bit old style :)
And too much OpenTX copy of something which could be improved

**Comment by Jedsters** [comment]
ok, that's tricky as a high priority beep, alarm etc., is typically going to be a simple sound so I'm not sure what you are after and I suspect it needs to be a simple clean sound so it can be heard above other noise (wind etc.). 
However a google of 
'download sounds beep chirp'
produces a number of websites which may be of use including some which appear to be royalty free / free to use etc.  such as
https://pixabay.com/sound-effects/search/beeps/
https://www.soundjay.com/beep-sounds-1.html
https://mixkit.co/free-sound-effects/beep/
etc.  (not checked for royalties / terms of use on the above)
But I suspect in the absence of anything else most would be happy with old-school.
If any of the above could be of use but need some mods (e.g. rather than a single beep you want 2 / 3 beeps etc. let me know, I don't mind doing a bit of editing - but I'll need pointers from someone.

HTH?

**Comment by agibson2** [comment]
So this issue is limited to only missing sounds and not that we would like the system tones as a special function (that lua has) that don't queue up against already playing wav files.  Is that a correct assumption?  Should we create a new issue for implementing the tones that lua has as a special function?

**Comment by bsongis-frsky** [comment]
The issue is about "Cheep, Ratata, Tick, Siren, Ring, SciFi, Robot, Chirp, Tada, Crickt, AlarmClk..."
What exactly is needed? 
- if fancy sounds, I prefer wav files over those old style sounds
- it just beep (with a variable length / main frequency / pattern like haptic), the function is already there, and could be added to PlayAudio

**Comment by agibson2** [comment]
All of the examples given in opentx are not wav files from what I understand on the opentx website (Beep1/2/3, Warn1/2, Cheep, Ratata, Tick, Siren, Ring, SciFi, Robot, Chirp, Tada, Crickt, AlarmClk.).  They seem to be generated in realtime as tones from what I understand.  The opentx site mentions that they are "Generating tones/beeps of different frequencies and durations".  I was hoping we could have the same in Ethos with similar feature where they are not queued by  wav files and played immediately.

**Comment by Jedsters** [comment]
I agree with @agibson2 what I'd like is a variety of high priority interrupts (similar to opentx) which are played immediately and not queued. I'd suggest some which are more informational (simple beeps), some which are more minor alerts, some more major alerts and some which are alarms. I think the opentx ones cover this. 

In terms of .wav / .mp3 files I suspect we all have our own thoughts as to what we want and can add those ourselves - although if Bertrand has some he'd like / more tuneful and modern, I'd welcome those being added to Ethos too - one can never have too much choice (well, maybe you can). I don't think there would be any functional difference in whether these are built into Ethos or user added.

I'm assuming if they were added as play audio this type of alerts these new sounds would be played immediately rather than queued? 
If these were called 'Play Sound' I guess all sounds would be interrupts. 
Thanks

**Comment by agibson2** [comment]
We don't know what is possible.  I think if wav files could be played at the same time as others, they could work instead of the system generated tones.  Maybe they could be called system tones/audio.  I have assumed that loading 2 audio files at the same time could be problematic which is why I suggested using system generated tones for them like OpenTx does.

**Comment by Jedsters** [comment]
I'm guessing that if we want these played immediately then there is going to have to be another category, i.e. Play Audio for .wavs and Values and then perhaps Play Alerts which don't have all the same ordering and stuff, but are just played immediately an SF become true.
A variety of variable pitch / pattern tones as suggested by Bertrand would suffice for me, I think I used Chirp to remind me I was in Calibrate mode (now done with a .wav and not a high priority sound anyway), but other than that it was simple beeps to warn me of something. 
Thanks
